jenkins/spark-nightly-build.sh

```diff
@@ -19,6 +19,11 @@ set -ex
 
 SCALA_BINARY_VER=${SCALA_BINARY_VER:-"2.12"}
 if [ $SCALA_BINARY_VER == "2.13" ]; then
+    # Run scala2.13 build and test against JDK17
+    export JAVA_HOME=$(echo /usr/lib/jvm/java-1.17.0-*)
+    update-java-alternatives --set $JAVA_HOME
+    java -version
```

Comment:
It deserves a separate issue. Just noting that it seems some of the repro difficulties with CI may stem from the fact that we constantly mutate PATH to activate different default JDK. Normally setting JAVA_HOME should be sufficient. 

However, if the project transitions to using toolchains https://maven.apache.org/plugins/maven-toolchains-plugin/toolchain-mojo.html#toolchains this does not work correctly.